### PR TITLE
tailscale: add acceptance test for device_authorization

### PR DESCRIPTION
**Pull request by mpminardi**
Add basic acceptance test for device_authorization. Also add logic to allow for importing `device_authorization` while I was here.

Updates https://github.com/tailscale/corp/issues/21842

### CODE REVIEW

File: tailscale/resource_device_authorization.go [hunk 1]
```diff
@@ -16,6 +16,9 @@ func resourceDeviceAuthorization() *schema.Resource {
 		CreateContext: resourceDeviceAuthorizationCreate,
 		UpdateContext: resourceDeviceAuthorizationUpdate,
 		DeleteContext: resourceDeviceAuthorizationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
```
Comment (mpminardi) [inline tailscale/resource_device_authorization.go:20]: Added ability to import this resource while I was here.

File: tailscale/resource_device_authorization.go [hunk 2]
```diff
@@ -33,7 +36,7 @@ func resourceDeviceAuthorization() *schema.Resource {
 
 func resourceDeviceAuthorizationRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	client := m.(*tailscale.Client)
-	deviceID := d.Get("device_id").(string)
+	deviceID := d.Id()
```
Comment (mpminardi) [inline tailscale/resource_device_authorization.go:39]: This is required for import to function as the `device_id` is not set when we are importing, although they are equivalent values.

File: tailscale/resource_device_authorization.go [hunk 3]
```diff
@@ -55,6 +58,7 @@ func resourceDeviceAuthorizationRead(ctx context.Context, d *schema.ResourceData
 	}
 
 	d.SetId(selected.ID)
+	d.Set("device_id", selected.ID)
```
Comment (mpminardi) [inline tailscale/resource_device_authorization.go:61]: Explicitly set the `device_id` as well for the import case / to prevent drift.

**Comment by oxtoacart** [comment]
LOL, I built a version of this as part of my V2 API work 🤣 

https://github.com/tailscale/terraform-provider-tailscale/pull/389

**Comment by mpminardi** [comment]
> LOL, I built a version of this as part of my V2 API work 🤣
> 
> #389

Whoops 🤾‍♂️

**Pull request merged**